src/platform/qt/scripting/AutorunScriptModel.cpp

```diff
@@ -5,22 +5,59 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 #include "scripting/AutorunScriptModel.h"
 
-#include "ConfigController.h"
-#include "LogController.h"
+#include <QtDebug>
+#include <iostream>
+
+QDataStream& operator<<(QDataStream& stream, const QGBA::AutorunScriptModel::ScriptInfo& object) {
+	stream << QGBA::AutorunScriptModel::ScriptInfo::VERSION;
+	stream << object.filename.toUtf8();
+	stream << object.active;
+	return stream;
+}
+
+QDataStream& operator>>(QDataStream& stream, QGBA::AutorunScriptModel::ScriptInfo& object) {
+	uint16_t version = 0;
+	stream >> version;
+	if (version == 1) {
+		QByteArray filename;
+		stream >> filename;
+		object.filename = QString::fromUtf8(filename);
+	} else {
+		qCritical() << QGBA::AutorunScriptModel::tr("Could not load autorun script settings: unknown script info format %1").arg(version);
+		stream.abortTransaction();
+		return stream;
+	}
+	stream >> object.active;
+	return stream;
+}
 
 using namespace QGBA;
 
-AutorunScriptModel::AutorunScriptModel(ConfigController* config, QObject* parent)
+void AutorunScriptModel::registerMetaTypes() {
+	qRegisterMetaType<AutorunScriptModel::ScriptInfo>();
+#if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
+	qRegisterMetaTypeStreamOperators<AutorunScriptModel::ScriptInfo>("QGBA::AutorunScriptModel::ScriptInfo");
+#endif
+}
+
+AutorunScriptModel::AutorunScriptModel(QObject* parent)
 	: QAbstractListModel(parent)
-	, m_config(config)
 {
-	QList<QVariant> autorun = m_config->getList("autorunSettings");
-	for (const auto& item: autorun) {
+	// Nothing to do
+}
+
+void AutorunScriptModel::deserialize(const QList<QVariant>& autorun) {
+	for (const auto& item : autorun) {
 		if (!item.canConvert<ScriptInfo>()) {
 			continue;
 		}
+		ScriptInfo info = qvariant_cast<ScriptInfo>(item);
+		if (info.filename.isEmpty()) {
+			continue;
+		}
```

Comment:
Added this check in here because `canConvert` doesn't detect if the variant will break during deserialization.